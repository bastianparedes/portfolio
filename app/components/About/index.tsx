import React from 'react';

import Image from 'next/image';

import images from './skillImages';
import styles from './styles.module.scss';
import constants from '../../../config/constants';
import { Section } from '../Common';

const About = (): JSX.Element => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2>{constants.ABOUT.DESCRIPTION_PART_1}</h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              rel="noreferrer noopener"
              target="_blank"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {Object.keys(images).map((skill) => (
              <div className={styles.skillCard} key={skill}>
                <div className={styles.skillImageContainer}>
                  <Image
                    alt={skill}
                    placeholder="blur"
                    src={images[skill as keyof typeof images]}
                    style={{
                      height: '100%',
                      objectFit: 'contain',
                      width: '100%'
                    }}
                  />
                </div>
                <p className={styles.skillName}>{skill}</p>
              </div>
            ))}
          </div>
        </div>
      </div>
    </Section>
  );
};

export default About;
