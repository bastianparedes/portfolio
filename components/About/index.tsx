import React from 'react'
import styles from './styles.module.css'
import { Section } from '../Common'
import constants from '../../config/constants.json'
import AboutSkill from './AboutSkill'

const About = (): JSX.Element => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2 className={styles.mainLeft__h2}>
            {constants.ABOUT.DESCRIPTION_PART_1}
          </h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              target="_blank"
              rel="noreferrer noopener"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {constants.ABOUT.SKILLS.LIST.map((skill) => (
              <AboutSkill key={skill.NAME} src={skill.SRC} name={skill.NAME} />
            ))}
          </div>
        </div>
      </div>
    </Section>
  )
}

export default About
