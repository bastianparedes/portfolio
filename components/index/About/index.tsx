import React from 'react';

import constants from '../../../config/constants';
import { Section } from '../Common';
import AboutSkill from './AboutSkill';
import styles from './styles.module.scss';

const About = (): JSX.Element => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2>{constants.ABOUT.DESCRIPTION_PART_1}</h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              target="_blank"
              rel="noreferrer noopener"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {constants.ABOUT.SKILLS.LIST.map((skill) => (
              <AboutSkill key={skill.NAME} src={skill.SRC} name={skill.NAME} />
            ))}
          </div>
        </div>
      </div>
    </Section>
  );
};

export default About;
