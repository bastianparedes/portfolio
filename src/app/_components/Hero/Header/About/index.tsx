'use client';

import { useState } from 'react';
import Modal from '../../../Modal';
import { useTranslationContext } from '../../../../_contexts/translation';

const Component = () => {
  const [showContent, setShowContent] = useState(false);
  const { translation } = useTranslationContext();
  const openModal = () => setShowContent(true);
  const closeModal = () => setShowContent(false);

  const yearsExperience = new Date().getFullYear() - 2020;

  return (
    <>
      <button onClick={openModal}>{translation.about.button}</button>
      {showContent && (
        <Modal closeModal={closeModal}>
          <div className="flex flex-col gap-5">
            <p>
              {translation.about.paragraph_1.part_1} {yearsExperience}{' '}
              {translation.about.paragraph_1.part_2}
            </p>
            <p>{translation.about.paragraph_2}</p>
          </div>
        </Modal>
      )}
    </>
  );
};

export default Component;
