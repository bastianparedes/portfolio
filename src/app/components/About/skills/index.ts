import css from './css.png';
import docker from './docker.png';
import drizzle from './drizzle-orm.png';
import git from './git.png';
import graphql from './graphql.png';
import html from './html5.png';
import java from './java.png';
import javascript from './javascript.png';
import mongodb from './mongodb.png';
import mysql from './mysql.png';
import next from './next.js.png';
import node from './nodejs.png';
import postgre from './postgresql.png';
import prisma from './prisma.png';
import python from './python3.png';
import react from './react.png';
import sass from './sass.png';
import typescript from './typescript.png';

const skills = [
  {
    image: css,
    name: 'CSS'
  },
  {
    image: docker,
    name: 'Docker'
  },
  {
    image: drizzle,
    name: 'Drizzle ORM'
  },
  {
    image: git,
    name: 'Git'
  },
  {
    image: graphql,
    name: 'GraphQL'
  },
  {
    image: html,
    name: 'HTML'
  },
  {
    image: java,
    name: 'Java'
  },
  {
    image: javascript,
    name: 'JavaScript'
  },
  {
    image: mongodb,
    name: 'Mongo DB'
  },
  {
    image: mysql,
    name: 'MySql'
  },
  {
    image: next,
    name: 'NextJS'
  },
  {
    image: node,
    name: 'Node'
  },
  {
    image: postgre,
    name: 'Postgre SQL'
  },
  {
    image: prisma,
    name: 'Prisma'
  },
  {
    image: python,
    name: 'Python'
  },
  {
    image: react,
    name: 'React'
  },
  {
    image: sass,
    name: 'Sass'
  },
  {
    image: typescript,
    name: 'TypeScript'
  }
];

export default skills;
