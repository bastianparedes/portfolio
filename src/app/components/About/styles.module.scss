@import 'bastianparedes/styles/_breakpoints';

.main {
  display: grid;
  gap: 80px;
  grid-template-columns: 1fr 1fr;
}

.mainLeft {
  &__p {
    text-align: justify;
    line-height: 40px;
    font-size: 1.2rem;
  }
}

.buttonCvContainer {
  display: flex;

  .buttonCv {
    font-size: 1.5rem;
    margin: 0 auto;
    padding: 1rem;
    background-color: rgb(103, 212, 255);
    border: black 1px solid;
    text-decoration: none;
    color: rgb(0, 0, 0);
  }
}

.mainRight {
  &__h2 {
    text-align: center;
  }
}

.skillsContainer {
  position: relative;
  padding-top: 0.9rem;
  display: grid;
  place-items: center;
  gap: 2.5rem 6rem;
  grid-template-columns: repeat(auto-fit, minmax(4rem, 1fr));
}

@include phone-and-tablet {
  .main {
    grid-template-columns: 1fr;
  }
  .skillsContainer {
    gap: 2.5rem 2.5rem;
    grid-template-columns: repeat(auto-fit, minmax(5rem, 1fr));
  }
}

@include phone {
  .skillsContainer {
    gap: 2.5rem 1rem;
    grid-template-columns: repeat(auto-fit, minmax(5rem, 1fr));
  }
}




.skillCard {
  width: 5rem;
  position: relative;
}

.skillImageContainer {
  width: 100%;
  height: 5rem;
  position: relative;
  margin-bottom: 0.5rem;
}

.skillName {
  font-weight: bolder;
  text-align: center;
  margin: 0;
}

@include phone-and-tablet {
  .skillCard {
    width: 5rem;
  }
  .skillName {
    font-size: 1rem;
  }
}

@include phone {
  .skillCard {
    width: 4rem;
  }
  .skillName {
    font-size: 0.8rem;
  }
}
