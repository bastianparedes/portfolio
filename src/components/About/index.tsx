import React from 'react'
import styles from './styles.module.css'
import { Section } from '../Common'

interface SkillProps {
  src: string
  name: string
}

const Skill = (props: SkillProps): JSX.Element => {
  return (
    <div className={styles.skillCard}>
      <img
        className={styles.skillImage}
        src={process.env.PUBLIC_URL + '/images/skills/' + props.src}
        alt={props.name}
      />
      <p className={styles.skillName}>{props.name}</p>
    </div>
  )
}

const About = (): JSX.Element => {
  return (
    <Section id="about" title="About me">
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2 className={styles.mainLeft__h2}>Soy Bastián Paredes,</h2>
          <p className={styles.mainLeft__p}>
            Un Front End Developer con experiencia construyendo sitios web con
            responsive design que conecten front-end y back-end. Me especializo
            en el uso de JavaScript, React.js y Next.js.
          </p>
          <p className={styles.mainLeft__p}>
            Siempre busco mejorar mis habilidades desarrollando proyectos
            propios a la par de mi trabajo. Estoy constantemente estudiando
            acerca de las nuevas tecnologías para actualizar mi repertorio de
            herramientas para que los códigos de mis creaciones sean fáciles de
            leer, seguros y eficientes.
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href="https://drive.google.com/file/d/1LheHpim4QsBKLe6kxYD1OMggrRW8DLWl/view?usp=sharing"
              target="_blank"
              rel="noreferrer noopener"
            >
              Ver currículum
            </a>
          </button>
        </div>
        <div className={styles.mainRight}>
          <h2 className={styles.mainRight__h2}>Skills</h2>
          <div className={styles.skillsContainer}>
            <Skill src="html5.png" name="HTML 5" />
            <Skill src="css3.png" name="CSS 3" />
            <Skill src="javascript.png" name="JavaScript" />
            <Skill src="typescript.png" name="TypeScript" />
            <Skill src="react.js.png" name="React.js" />
            <Skill src="next.js.png" name="Next.js" />
            <Skill src="python3.png" name="Python 3" />
            <Skill src="mysql.png" name="MySql" />
            <Skill src="mongodb.svg" name="MongoDB" />
          </div>
        </div>
      </div>
    </Section>
  )
}

export default About
