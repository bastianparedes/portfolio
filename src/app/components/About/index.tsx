import React from 'react';

import Image from 'next/image';

import skills from './skills';
import styles from './styles.module.scss';
import constants from '../../../../config/constants';
import { Section } from '../Common';

const About = () => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2>{constants.ABOUT.DESCRIPTION_PART_1}</h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              rel="noreferrer noopener"
              target="_blank"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {skills.map((skill) => (
              <div className={styles.skillCard} key={skill.name}>
                <div className={styles.skillImageContainer}>
                  <Image
                    alt={skill.name}
                    placeholder="blur"
                    src={skill.image}
                    style={{
                      height: '100%',
                      objectFit: 'contain',
                      width: '100%'
                    }}
                  />
                </div>
                <p className={styles.skillName}>{skill.name}</p>
              </div>
            ))}
          </div>
        </div>
      </div>
    </Section>
  );
};

export default About;
