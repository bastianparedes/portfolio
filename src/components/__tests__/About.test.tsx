import React from 'react'
import About from '../About'
import { render } from '@testing-library/react'

describe('<About />', () => {
  it('should render', () => {
    const { container } = render(<About />)
    expect(container).toMatchSnapshot()
  })
})
