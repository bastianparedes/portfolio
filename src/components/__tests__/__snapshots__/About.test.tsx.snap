// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<About /> should render 1'] = `
<div>
  <section
    class="section"
    id="about"
  >
    <h3
      class="title"
    >
      About me
    </h3>
    <div
      class="main"
    >
      <div
        class="mainLeft"
      >
        <h2
          class="mainLeft__h2"
        >
          Soy Bastián Paredes,
        </h2>
        <p
          class="mainLeft__p"
        >
          Un Front End Developer con experiencia construyendo sitios web con responsive design que conecten front-end y back-end. Me especializo en el uso de JavaScript, React.js y Next.js.
        </p>
        <p
          class="mainLeft__p"
        >
          Siempre busco mejorar mis habilidades desarrollando proyectos propios a la par de mi trabajo. Estoy constantemente estudiando acerca de las nuevas tecnologías para actualizar mi repertorio de herramientas para que los códigos de mis creaciones sean fáciles de leer, seguros y eficientes.
        </p>
        <button
          class="buttonCv"
        >
          <a
            class="buttonCv__a"
            href="https://drive.google.com/file/d/1LheHpim4QsBKLe6kxYD1OMggrRW8DLWl/view?usp=sharing"
            rel="noreferrer noopener"
            target="_blank"
          >
            Ver currículum
          </a>
        </button>
      </div>
      <div
        class="mainRight"
      >
        <h2
          class="mainRight__h2"
        >
          Skills
        </h2>
        <div
          class="skillsContainer"
        >
          <div
            class="skillCard"
          >
            <img
              alt="HTML 5"
              class="skillImage"
              src="/images/skills/html5.png"
            />
            <p
              class="skillName"
            >
              HTML 5
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="CSS 3"
              class="skillImage"
              src="/images/skills/css3.png"
            />
            <p
              class="skillName"
            >
              CSS 3
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="JavaScript"
              class="skillImage"
              src="/images/skills/javascript.png"
            />
            <p
              class="skillName"
            >
              JavaScript
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="TypeScript"
              class="skillImage"
              src="/images/skills/typescript.png"
            />
            <p
              class="skillName"
            >
              TypeScript
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="React.js"
              class="skillImage"
              src="/images/skills/react.js.png"
            />
            <p
              class="skillName"
            >
              React.js
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="Next.js"
              class="skillImage"
              src="/images/skills/next.js.png"
            />
            <p
              class="skillName"
            >
              Next.js
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="Python 3"
              class="skillImage"
              src="/images/skills/python3.png"
            />
            <p
              class="skillName"
            >
              Python 3
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="MySql"
              class="skillImage"
              src="/images/skills/mysql.png"
            />
            <p
              class="skillName"
            >
              MySql
            </p>
          </div>
          <div
            class="skillCard"
          >
            <img
              alt="MongoDB"
              class="skillImage"
              src="/images/skills/mongodb.svg"
            />
            <p
              class="skillName"
            >
              MongoDB
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
